### PR TITLE
Add new feature page for ovirt-lldp-labeler

### PR DESCRIPTION
**Pull request by almusil**
I confirm that this pull request was submitted according to the [contribution guidelines](https://github.com/oVirt/ovirt-site/blob/master/CONTRIBUTING.md): @almusil

This pull request needs review by: @dominikholler  @dankenigsberg @mburman5

### CODE REVIEW

File: source/develop/release-management/features/network/ovirt-lldp-labeler.html.md [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+feature_modules: ovirt-lldp-labeler
+feature_status: In Development
+---
+
+# oVirt LLDP labeler
+
+## Summary
+
+oVirt LLDP labeler is tool that utilize [LLDP support](https://ovirt.org/blog/2017/11/webadmin-lldp/) 
+added to oVirt in 4.2. The tool helps to some extent automate network overhead
```
Comment (dankenigsberg) [inline source/develop/release-management/features/network/ovirt-lldp-labeler.html.md]: network -> network configuration

File: source/develop/release-management/features/network/ovirt-lldp-labeler.html.md [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+feature_status: In Development
+---
+
+# oVirt LLDP labeler
+
+## Summary
+
+oVirt LLDP labeler is tool that utilize [LLDP support](https://ovirt.org/blog/2017/11/webadmin-lldp/) 
+added to oVirt in 4.2. The tool helps to some extent automate network overhead 
+over clusters that are managed by the labeler. It can be used as separate
```
Comment (dankenigsberg) [inline source/develop/release-management/features/network/ovirt-lldp-labeler.html.md]: over->in

File: source/develop/release-management/features/network/ovirt-lldp-labeler.html.md [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+feature_name: oVirt LLDP labeler
+feature_modules: ovirt-lldp-labeler
+feature_status: In Development
+---
+
+# oVirt LLDP labeler
+
+## Summary
+
+oVirt LLDP labeler is tool that utilize [LLDP support](https://ovirt.org/blog/2017/11/webadmin-lldp/)
```
[the remark this answers is not in the document]
Comment (almusil) [reply source/develop/release-management/features/network/ovirt-lldp-labeler.html.md]: Done

File: source/develop/release-management/features/network/ovirt-lldp-labeler.html.md (hunk 1 above)
Comment (almusil) [reply source/develop/release-management/features/network/ovirt-lldp-labeler.html.md]: Done

File: source/develop/release-management/features/network/ovirt-lldp-labeler.html.md [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+* `api_url` URL of the engine API
+* `ca_file` Path to certficate that should be used for API access
+* `auto_bonding` Switch to enable/disable auto bonding
+* `auto_labeling` Switch to enable/disable auto labeling
+
+The last configuration file is configuration of the systemd timer. It is usually
+located in `/usr/lib/systemd/system/` and it is called 
+`ovirt-lldp-labeler.timer`. 
+* `OnUnitActiveSec` The timer delay for single labeler runs.
+
```
[the remark this answers is not in the document]
Comment (almusil) [reply source/develop/release-management/features/network/ovirt-lldp-labeler.html.md]: Done

File: source/develop/release-management/features/network/ovirt-lldp-labeler.html.md [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+## Prerequisites
+
+The labler has to be capable of reaching desired engine REST API and the 
+`python-ovirt-engine-sdk4` in version greater than `4.2` has to be installed.
+
+
+## Limitations
+
+Currently only the Juniper switches supports reporting of the needed parameters.
+
```
[the remark this answers is not in the document]
Comment (almusil) [reply source/develop/release-management/features/network/ovirt-lldp-labeler.html.md]: Done

File: source/develop/release-management/features/network/ovirt-lldp-labeler.html.md [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+## Testing
+
+* Labeler with auto bonding disabled
+* Labeler with auto labeling disabled
+* Auto bonding and labeling enabled at the same time
+* Hosts with vlan interfaces
+* Hosts with interfaces with disabled LLDP reporting
+* Installation
+* Security - only the ovirt-lldp-credentials.conf should be able to read/write 
+into the config
```
[the remark this answers is not in the document]
Comment (almusil) [reply source/develop/release-management/features/network/ovirt-lldp-labeler.html.md]: I meant to write: owner should be

File: source/develop/release-management/features/network/ovirt-lldp-labeler.html.md [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+`python-ovirt-engine-sdk4` in version greater than `4.2` has to be installed.
+
+
+## Limitations
+
+Currently only the Juniper switches supports reporting of the needed parameters.
+
+
+## Implementation
+
```
[the remark this answers is not in the document]
Comment (almusil) [reply source/develop/release-management/features/network/ovirt-lldp-labeler.html.md]: Good idea, done, thanks

File: source/develop/release-management/features/network/ovirt-lldp-labeler.html.md [hunk 8]
```diff
@@ -0,0 +1,127 @@
...
+feature_name: oVirt LLDP labeler
+feature_modules: ovirt-lldp-labeler
+feature_status: In Development
+---
+
+# oVirt LLDP labeler
+
+## Summary
+
+oVirt LLDP labeler is tool that utilizes [LLDP support](https://ovirt.org/blog/2017/11/webadmin-lldp/)
```
Comment (dominikholler) [inline source/develop/release-management/features/network/ovirt-lldp-labeler.html.md]: is a tool

File: source/develop/release-management/features/network/ovirt-lldp-labeler.html.md [hunk 9]
```diff
@@ -0,0 +1,127 @@
...
+---
+
+# oVirt LLDP labeler
+
+## Summary
+
+oVirt LLDP labeler is tool that utilizes [LLDP support](https://ovirt.org/blog/2017/11/webadmin-lldp/) 
+added to oVirt in 4.2. The tool helps to some extent automate network 
+configuration overhead in clusters that are managed by the labeler. 
+It can be used as separate service or as one time tool using its cli.
```
Comment (dominikholler) [inline source/develop/release-management/features/network/ovirt-lldp-labeler.html.md]: as a separate

Comment (dominikholler) [inline source/develop/release-management/features/network/ovirt-lldp-labeler.html.md]: one-time

File: source/develop/release-management/features/network/ovirt-lldp-labeler.html.md [hunk 10]
```diff
@@ -0,0 +1,127 @@
...
+- Feature Owner: Ales Musil (amusil)
+- E-mail: <amusil@redhat.com>
+
+
+### Benefit to oVirt
+
+The labeler is service that is capable of helping oVirt administrator
+to automate some network related tasks. According to configuration reported 
+via LLDP from peer switch, the labaler is capable of labeling host network 
+interface or bond interfaces together.
```
Comment (dominikholler) [inline source/develop/release-management/features/network/ovirt-lldp-labeler.html.md]: labaler -> labeler

File: source/develop/release-management/features/network/ovirt-lldp-labeler.html.md [hunk 11]
```diff
@@ -0,0 +1,127 @@
...
+### Benefit to oVirt
+
+The labeler is service that is capable of helping oVirt administrator
+to automate some network related tasks. According to configuration reported 
+via LLDP from peer switch, the labaler is capable of labeling host network 
+interface or bond interfaces together. 
+
+## Prerequisites
+
+The labler has to be capable of reaching desired engine REST API and the
```
Comment (dominikholler) [inline source/develop/release-management/features/network/ovirt-lldp-labeler.html.md]: labler -> labeler

File: source/develop/release-management/features/network/ovirt-lldp-labeler.html.md [hunk 12]
```diff
@@ -0,0 +1,127 @@
...
+
+## Prerequisites
+
+The labler has to be capable of reaching desired engine REST API and the 
+`python-ovirt-engine-sdk4` in version greater than `4.2` has to be installed.
+
+
+## Limitations
+
+Currently only the Juniper switches supports reporting of the needed parameters.
```
Comment (dominikholler) [inline source/develop/release-management/features/network/ovirt-lldp-labeler.html.md]: Currently, only Juniper switches support

File: source/develop/release-management/features/network/ovirt-lldp-labeler.html.md [hunk 13]
```diff
@@ -0,0 +1,127 @@
...
+
+## Implementation
+
+Overall architecture:
+![](/images/wiki/ovirt-lldp-labeler-architecture.png)
+
+### Service
+
+The labeler service is done through timers availble in systemd. The timer calls
+target which executes the cli tool. The delay is one hour by default and can be
```
Comment (dominikholler) [inline source/develop/release-management/features/network/ovirt-lldp-labeler.html.md]: availble -> available

File: source/develop/release-management/features/network/ovirt-lldp-labeler.html.md [hunk 14]
```diff
@@ -0,0 +1,127 @@
...
+
+The labeler service is done through timers availble in systemd. The timer calls
+target which executes the cli tool. The delay is one hour by default and can be 
+changed in the timer specification.
+
+### Labeler
+
+The tool itself is divided into two parts. The labeling part and the bonding 
+part, each can be enabled or disabled in configuration file. 
+
```
Comment (dominikholler) [inline source/develop/release-management/features/network/ovirt-lldp-labeler.html.md]: in the configuration file

File: source/develop/release-management/features/network/ovirt-lldp-labeler.html.md [hunk 15]
```diff
@@ -0,0 +1,127 @@
...
+Each step described below is done for each host, that is part of the cluster 
+specified in the configuration file.
+
+- The bonding part starts by fetching all LLDP availble for interfaces that are 
+not part of any existing bond already. The labeler is not trying to modify
```
Comment (dominikholler) [inline source/develop/release-management/features/network/ovirt-lldp-labeler.html.md]: availble -> available

File: source/develop/release-management/features/network/ovirt-lldp-labeler.html.md [hunk 16]
```diff
@@ -0,0 +1,127 @@
...
+part, each can be enabled or disabled in configuration file. 
+
+Each step described below is done for each host, that is part of the cluster 
+specified in the configuration file.
+
+- The bonding part starts by fetching all LLDP availble for interfaces that are 
+not part of any existing bond already. The labeler is not trying to modify 
+existing bonds, just to create new ones. The LLDP information is then searched 
+for `Link Aggregation` capabilites. The interfaces, that are capable of 
+aggregation, are grouped together by `Aggregation Link ID`. Before creating
```
Comment (dominikholler) [inline source/develop/release-management/features/network/ovirt-lldp-labeler.html.md]: capabilites -> capabilities

File: source/develop/release-management/features/network/ovirt-lldp-labeler.html.md [hunk 17]
```diff
@@ -0,0 +1,127 @@
...
+
+Each step described below is done for each host, that is part of the cluster 
+specified in the configuration file.
+
+- The bonding part starts by fetching all LLDP availble for interfaces that are 
+not part of any existing bond already. The labeler is not trying to modify 
+existing bonds, just to create new ones. The LLDP information is then searched 
+for `Link Aggregation` capabilites. The interfaces, that are capable of 
+aggregation, are grouped together by `Aggregation Link ID`. Before creating 
+bond, the interface with attached `ovirtmgmt` network is filtered out, as we
```
Comment (dominikholler) [inline source/develop/release-management/features/network/ovirt-lldp-labeler.html.md]: "a bond" or "the bond"

File: source/develop/release-management/features/network/ovirt-lldp-labeler.html.md [hunk 18]
```diff
@@ -0,0 +1,127 @@
...
+specified in the configuration file.
+
+- The bonding part starts by fetching all LLDP availble for interfaces that are 
+not part of any existing bond already. The labeler is not trying to modify 
+existing bonds, just to create new ones. The LLDP information is then searched 
+for `Link Aggregation` capabilites. The interfaces, that are capable of 
+aggregation, are grouped together by `Aggregation Link ID`. Before creating 
+bond, the interface with attached `ovirtmgmt` network is filtered out, as we 
+don't want to risk losing connectivity with the host. Also rules about which 
+networks can be attached to interface are applied. After creating bond
```
Comment (dominikholler) [inline source/develop/release-management/features/network/ovirt-lldp-labeler.html.md]: "an interface"

File: source/develop/release-management/features/network/ovirt-lldp-labeler.html.md [hunk 19]
```diff
@@ -0,0 +1,127 @@
...
+Each step described below is done for each host, that is part of the cluster 
+specified in the configuration file.
+
+- The bonding part starts by fetching all LLDP availble for interfaces that are 
+not part of any existing bond already. The labeler is not trying to modify 
+existing bonds, just to create new ones. The LLDP information is then searched 
+for `Link Aggregation` capabilites. The interfaces, that are capable of 
+aggregation, are grouped together by `Aggregation Link ID`. Before creating 
+bond, the interface with attached `ovirtmgmt` network is filtered out, as we 
+don't want to risk losing connectivity with the host. Also rules about which
```
Comment (dominikholler) [inline source/develop/release-management/features/network/ovirt-lldp-labeler.html.md]: Also,

File: source/develop/release-management/features/network/ovirt-lldp-labeler.html.md [hunk 20]
```diff
@@ -0,0 +1,127 @@
...
+- The bonding part starts by fetching all LLDP availble for interfaces that are 
+not part of any existing bond already. The labeler is not trying to modify 
+existing bonds, just to create new ones. The LLDP information is then searched 
+for `Link Aggregation` capabilites. The interfaces, that are capable of 
+aggregation, are grouped together by `Aggregation Link ID`. Before creating 
+bond, the interface with attached `ovirtmgmt` network is filtered out, as we 
+don't want to risk losing connectivity with the host. Also rules about which 
+networks can be attached to interface are applied. After creating bond 
+candidates the last step is to re-attach all networks from slaves to bond 
+candidate. All network configurations made by the labeler are marked as
```
Comment (dominikholler) [inline source/develop/release-management/features/network/ovirt-lldp-labeler.html.md]: the bond candidate

File: source/develop/release-management/features/network/ovirt-lldp-labeler.html.md [hunk 21]
```diff
@@ -0,0 +1,127 @@
...
+candidate. All network configurations made by the labeler are marked as 
+permanent so host restart won't wipe them.
+
+
+- The labeling part fetches LLDP from all non-vlan interfaces and bond slaves.
+The labeler search every interface LLDP information for `VLAN ID` tlvs. For 
+every VLAN tlv, the labeler creates label candidate, in format of 
+`lldp_vlan_$(VLAN-ID)`, where `$(VLAN-ID)` is substituted with VLAN id from 
+the reported LLDP information. It is possible to have multiple VLAN on 
+the interface, in this case the interface ends up with multiple labels.
```
Comment (dominikholler) [inline source/develop/release-management/features/network/ovirt-lldp-labeler.html.md]: case,

File: source/develop/release-management/features/network/ovirt-lldp-labeler.html.md [hunk 22]
```diff
@@ -0,0 +1,127 @@
...
+aggregation, are grouped together by `Aggregation Link ID`. Before creating 
+bond, the interface with attached `ovirtmgmt` network is filtered out, as we 
+don't want to risk losing connectivity with the host. Also rules about which 
+networks can be attached to interface are applied. After creating bond 
+candidates the last step is to re-attach all networks from slaves to bond 
+candidate. All network configurations made by the labeler are marked as 
+permanent so host restart won't wipe them.
+
+
+- The labeling part fetches LLDP from all non-vlan interfaces and bond slaves.
```
Comment (dominikholler) [inline source/develop/release-management/features/network/ovirt-lldp-labeler.html.md]: vlan -> VLAN ?

File: source/develop/release-management/features/network/ovirt-lldp-labeler.html.md [hunk 23]
```diff
@@ -0,0 +1,127 @@
...
+the reported LLDP information. It is possible to have multiple VLAN on 
+the interface, in this case the interface ends up with multiple labels. 
+
+
+## Configuration
+
+The labeler consists of three main configuration files.
+Full configuration documentation is available in the [project readme](https://github.com/oVirt/ovirt-lldp-labeler/blob/master/README.adoc#configuration).
+
+First two files are usually located under `/etc/ovirt-lldp-labeler/conf.d/`.
```
Comment (dominikholler) [inline source/develop/release-management/features/network/ovirt-lldp-labeler.html.md]: under -> in ?

File: source/develop/release-management/features/network/ovirt-lldp-labeler.html.md [hunk 24]
```diff
@@ -0,0 +1,127 @@
...
+
+The labeler consists of three main configuration files.
+Full configuration documentation is available in the [project readme](https://github.com/oVirt/ovirt-lldp-labeler/blob/master/README.adoc#configuration).
+
+First two files are usually located under `/etc/ovirt-lldp-labeler/conf.d/`. 
+
+The `ovirt-lldp-credentials.conf` contains login information to access the 
+engine API. This file is restricted with permission 0600 to prevent any
+unwanted access to the file.
+* `username` username for administrator of the engine
```
Comment (dominikholler) [inline source/develop/release-management/features/network/ovirt-lldp-labeler.html.md]: "an administrator",  but why is an admin account required?
maybe
"username for Engine"
is more simple?

File: source/develop/release-management/features/network/ovirt-lldp-labeler.html.md [hunk 25]
```diff
@@ -0,0 +1,127 @@
...
+* `password` password for administrator of the engine
```
Comment (dominikholler) [inline source/develop/release-management/features/network/ovirt-lldp-labeler.html.md]: same like username

File: source/develop/release-management/features/network/ovirt-lldp-labeler.html.md [hunk 26]
```diff
@@ -0,0 +1,127 @@
...
+
+
+The `ovirt-lldp-labeler.conf` contains basic configuration. 
+* `clusters` list of clusters on top of which the labeler will run
+* `api_url` URL of the engine API
+* `ca_file` Path to certficate that should be used for API access
```
Comment (dominikholler) [inline source/develop/release-management/features/network/ovirt-lldp-labeler.html.md]: Path -> path

File: source/develop/release-management/features/network/ovirt-lldp-labeler.html.md [hunk 27]
```diff
@@ -0,0 +1,127 @@
...
+* `auto_bonding` Switch to enable/disable auto bonding
+* `auto_labeling` Switch to enable/disable auto labeling
+
+The last configuration file is configuration of the systemd timer. It is usually
```
Comment (dominikholler) [inline source/develop/release-management/features/network/ovirt-lldp-labeler.html.md]: ... is the configuration...

File: source/develop/release-management/features/network/ovirt-lldp-labeler.html.md [hunk 28]
```diff
@@ -0,0 +1,127 @@
...
+* `OnUnitActiveSec` The timer delay for single labeler runs.
+
+
+
+## Testing
+
+* Labeler with auto bonding disabled
+* Labeler with auto labeling disabled
+* Auto bonding and labeling enabled at the same time
+* Hosts with vlan interfaces
```
Comment (dominikholler) [inline source/develop/release-management/features/network/ovirt-lldp-labeler.html.md]: vlan -> VLAN ?

File: source/develop/release-management/features/network/ovirt-lldp-labeler.html.md [hunk 29]
```diff
@@ -0,0 +1,127 @@
...
+
+The last configuration file is configuration of the systemd timer. It is usually
+located in `/usr/lib/systemd/system/` and it is called 
+`ovirt-lldp-labeler.timer`. 
+* `OnUnitActiveSec` The timer delay for single labeler runs.
+
+
+
+## Testing
+
```
Comment (dominikholler) [inline source/develop/release-management/features/network/ovirt-lldp-labeler.html.md]: Some more words describing what should be checked might be helpful.

**Comment by almusil** [comment]
Not sure with the flow yet how to properly describe it

File: source/develop/release-management/features/network/ovirt-lldp-labeler.html.md (hunk 29 above)
Comment (almusil) [reply source/develop/release-management/features/network/ovirt-lldp-labeler.html.md]: Done

File: source/develop/release-management/features/network/ovirt-lldp-labeler.html.md [hunk 30]
```diff
@@ -0,0 +1,155 @@
...
+enabled for the interfaces connected to the hosts. To automatically 
+connected our logical network `vlan10` with the `eth1`, on every host,
+the administrator just needs to configure and run the labeler:
+
+First time setup:
+
+1) Write username and password into `/etc/ovirt-lldp-labeler/conf.d/ovirt-lldp-credentials.conf`
+2) Write cluster name and API address into `/etc/ovirt-lldp-labeler/conf.d/ovirt-lldp-labeler.conf`
+
+Repeatable setup:
```
Comment (dominikholler) [inline source/develop/release-management/features/network/ovirt-lldp-labeler.html.md]: I would call this "Repeatable invocation:"

File: source/develop/release-management/features/network/ovirt-lldp-labeler.html.md [hunk 31]
```diff
@@ -0,0 +1,155 @@
...
+
+1) Run the oVirt LLDP labeler by executing `python /usr/share/ovirt-lldp-labeler/ovirt_lldp_labeler_cli.py`
+  
+After the labeler run the hosts will automatically connect their
+`eth1` with `vlan10` network, due to matching label. This can be
```
Comment (dominikholler) [inline source/develop/release-management/features/network/ovirt-lldp-labeler.html.md]: Please write on which entities which label is attached, and that the labeler is doing this.

File: source/develop/release-management/features/network/ovirt-lldp-labeler.html.md [hunk 32]
```diff
@@ -0,0 +1,155 @@
...
+2) Write cluster name and API address into `/etc/ovirt-lldp-labeler/conf.d/ovirt-lldp-labeler.conf`
+
+Repeatable setup:
+
+1) Run the oVirt LLDP labeler by executing `python /usr/share/ovirt-lldp-labeler/ovirt_lldp_labeler_cli.py`
+  
+After the labeler run the hosts will automatically connect their
+`eth1` with `vlan10` network, due to matching label. This can be 
+repeated for every new host in the cluster, as well as for network
+changes with current hosts e.g. the switch port connection was
```
Comment (dominikholler) [inline source/develop/release-management/features/network/ovirt-lldp-labeler.html.md]: repeated every time a new host is added to the cluster, as well as for network

File: source/develop/release-management/features/network/ovirt-lldp-labeler.html.md [hunk 33]
```diff
@@ -0,0 +1,155 @@
...
+
+### Benefit to oVirt
+
+The labeler is service that is capable of helping oVirt administrator
+to automate some network related tasks. According to configuration reported 
+via LLDP from peer switch, the labeler is capable of labeling host network 
+interface or bond interfaces together. 
+
+### Typical user flow
+
```
Comment (dominikholler) [inline source/develop/release-management/features/network/ovirt-lldp-labeler.html.md]: "Example VLAN Flow"

File: source/develop/release-management/features/network/ovirt-lldp-labeler.html.md [hunk 34]
```diff
@@ -0,0 +1,156 @@
...
+### Example VLAN flow
+
+Consider following scenario:
+
+Engine contains logical network `vlan10` tagged as VLAN with id 10
+and label `lldp_vlan_10`. Multiple hosts that have it's `eth1` 
+connected to switch with VLAN id 10 configured and LLDP reporting 
+enabled for the interfaces connected to the hosts. To automatically 
+connected our logical network `vlan10` with the `eth1`, on every host,
```
Comment (dominikholler) [inline source/develop/release-management/features/network/ovirt-lldp-labeler.html.md]: connected our logical network `vlan10` with the `eth1` on every host,
(no comma)

File: source/develop/release-management/features/network/ovirt-lldp-labeler.html.md [hunk 35]
```diff
@@ -0,0 +1,156 @@
...
+
+1) Write username and password into `/etc/ovirt-lldp-labeler/conf.d/ovirt-lldp-credentials.conf`
+2) Write cluster name and API address into `/etc/ovirt-lldp-labeler/conf.d/ovirt-lldp-labeler.conf`
+
+Repeatable invocation:
+
+1) Run the oVirt LLDP labeler by executing `python /usr/share/ovirt-lldp-labeler/ovirt_lldp_labeler_cli.py`
+  
+After the labeler run, the hosts will automatically sets `lldp_vlan_10` 
+label on top of the `eth1`, which results into automatic connection of
```
Comment (dominikholler) [inline source/develop/release-management/features/network/ovirt-lldp-labeler.html.md]: results in the automatic

File: source/develop/release-management/features/network/ovirt-lldp-labeler.html.md [hunk 36]
```diff
@@ -0,0 +1,156 @@
...
+`eth1` with `vlan10` network, due to matching label. This can be 
+repeated every time a new host is added to the cluster, as well
```
Comment (dominikholler) [inline source/develop/release-management/features/network/ovirt-lldp-labeler.html.md]: due to the matching

File: source/develop/release-management/features/network/ovirt-lldp-labeler.html.md [hunk 37]
```diff
@@ -0,0 +1,156 @@
...
+repeated every time a new host is added to the cluster, as well 
+as for network changes with current hosts e.g. the switch port 
+connection was switched between `eth1` and `eth2` on the host.
+
+
+## Prerequisites
+
+The labeler has to be capable of reaching desired engine REST API and the 
+`python-ovirt-engine-sdk4` in version greater than `4.2` has to be installed.
+
```
Comment (dominikholler) [inline source/develop/release-management/features/network/ovirt-lldp-labeler.html.md]: in a version

File: source/develop/release-management/features/network/ovirt-lldp-labeler.html.md [hunk 38]
```diff
@@ -0,0 +1,156 @@
...
+
+## Limitations
+
+Currently, only the Juniper switches supports reporting of the needed parameters.
+The required parameters are:
```
Comment (dominikholler) [inline source/develop/release-management/features/network/ovirt-lldp-labeler.html.md]: support (without the s in the end)

Comment (almusil) [reply source/develop/release-management/features/network/ovirt-lldp-labeler.html.md]: Done

File: source/develop/release-management/features/network/ovirt-lldp-labeler.html.md (hunk 37 above)
Comment (almusil) [reply source/develop/release-management/features/network/ovirt-lldp-labeler.html.md]: Done

File: source/develop/release-management/features/network/ovirt-lldp-labeler.html.md (hunk 36 above)
Comment (almusil) [reply source/develop/release-management/features/network/ovirt-lldp-labeler.html.md]: Done

File: source/develop/release-management/features/network/ovirt-lldp-labeler.html.md (hunk 35 above)
Comment (almusil) [reply source/develop/release-management/features/network/ovirt-lldp-labeler.html.md]: Done

File: source/develop/release-management/features/network/ovirt-lldp-labeler.html.md (hunk 34 above)
Comment (almusil) [reply source/develop/release-management/features/network/ovirt-lldp-labeler.html.md]: Done

File: source/develop/release-management/features/network/ovirt-lldp-labeler.html.md [hunk 39]
```diff
@@ -0,0 +1,156 @@
...
+1) Write username and password into `/etc/ovirt-lldp-labeler/conf.d/ovirt-lldp-credentials.conf`
+2) Write cluster name and API address into `/etc/ovirt-lldp-labeler/conf.d/ovirt-lldp-labeler.conf`
+
+Repeatable invocation:
+
+1) Run the oVirt LLDP labeler by executing `python /usr/share/ovirt-lldp-labeler/ovirt_lldp_labeler_cli.py`
+  
+After the labeler run, the hosts will automatically sets `lldp_vlan_10` 
+label on top of the `eth1`, which results in the automatic connection of
+`eth1` with `vlan10` network, due to the matching label. This can be
```
Comment (mmirecki) [inline source/develop/release-management/features/network/ovirt-lldp-labeler.html.md]: `vlan10` network ->  the `vlan10` network

File: source/develop/release-management/features/network/ovirt-lldp-labeler.html.md [hunk 40]
```diff
@@ -0,0 +1,156 @@
...
+
+Repeatable invocation:
+
+1) Run the oVirt LLDP labeler by executing `python /usr/share/ovirt-lldp-labeler/ovirt_lldp_labeler_cli.py`
+  
+After the labeler run, the hosts will automatically sets `lldp_vlan_10` 
+label on top of the `eth1`, which results in the automatic connection of
+`eth1` with `vlan10` network, due to the matching label. This can be 
+repeated every time a new host is added to the cluster, as well 
+as for network changes with current hosts e.g. the switch port
```
Comment (mmirecki) [inline source/develop/release-management/features/network/ovirt-lldp-labeler.html.md]: with -> on
OR
with-> within

File: source/develop/release-management/features/network/ovirt-lldp-labeler.html.md [hunk 41]
```diff
@@ -0,0 +1,156 @@
...
+label on top of the `eth1`, which results in the automatic connection of
+`eth1` with `vlan10` network, due to the matching label. This can be 
+repeated every time a new host is added to the cluster, as well 
+as for network changes with current hosts e.g. the switch port 
+connection was switched between `eth1` and `eth2` on the host.
+
+
+## Prerequisites
+
+The labeler has to be capable of reaching desired engine REST API and the
```
Comment (mmirecki) [inline source/develop/release-management/features/network/ovirt-lldp-labeler.html.md]: desired -> the  desired

File: source/develop/release-management/features/network/ovirt-lldp-labeler.html.md [hunk 42]
```diff
@@ -0,0 +1,156 @@
...
+
+## Prerequisites
+
+The labeler has to be capable of reaching desired engine REST API and the 
+`python-ovirt-engine-sdk4` in a version greater than `4.2` has to be installed.
+
+
+## Limitations
+
+Currently, only the Juniper switches support reporting of the needed parameters.
```
Comment (mmirecki) [inline source/develop/release-management/features/network/ovirt-lldp-labeler.html.md]: the Juniper -> Juniper

File: source/develop/release-management/features/network/ovirt-lldp-labeler.html.md [hunk 43]
```diff
@@ -0,0 +1,156 @@
...
+
+## Implementation
+
+Overall architecture:
+![](/images/wiki/ovirt-lldp-labeler-architecture.png)
+
+### Service
+
+The labeler service is done through timers available in systemd. The timer calls
+target which executes the cli tool. The delay is one hour by default and can be
```
Comment (mmirecki) [inline source/develop/release-management/features/network/ovirt-lldp-labeler.html.md]: target -> a target

File: source/develop/release-management/features/network/ovirt-lldp-labeler.html.md [hunk 44]
```diff
@@ -0,0 +1,156 @@
...
+### Service
+
+The labeler service is done through timers available in systemd. The timer calls
+target which executes the cli tool. The delay is one hour by default and can be 
+changed in the timer specification.
+
+### Labeler
+
+The tool itself is divided into two parts. The labeling part and the bonding 
+art, each can be enabled or disabled in the configuration file.
```
Comment (mmirecki) [inline source/develop/release-management/features/network/ovirt-lldp-labeler.html.md]: art -> part

File: source/develop/release-management/features/network/ovirt-lldp-labeler.html.md [hunk 45]
```diff
@@ -0,0 +1,156 @@
...
+
+### Owner
+
+- Feature Owner: Ales Musil (amusil)
+- E-mail: <amusil@redhat.com>
+
+
+### Benefit to oVirt
+
+The labeler is service that is capable of helping oVirt administrator
```
[the remark this answers is not in the document]
Comment (almusil) [reply source/develop/release-management/features/network/ovirt-lldp-labeler.html.md]: Done

[the remark this answers is not in the document]
Comment (almusil) [reply source/develop/release-management/features/network/ovirt-lldp-labeler.html.md]: Done

File: source/develop/release-management/features/network/ovirt-lldp-labeler.html.md [hunk 46]
```diff
@@ -0,0 +1,156 @@
...
+
+### Example VLAN flow
+
+Consider following scenario:
+
+Engine contains logical network `vlan10` tagged as VLAN with id 10
+and label `lldp_vlan_10`. Multiple hosts that have it's `eth1` 
+connected to switch with VLAN id 10 configured and LLDP reporting 
+enabled for the interfaces connected to the hosts. To automatically 
+connected our logical network `vlan10` with the `eth1` on every host,
```
[the remark this answers is not in the document]
Comment (almusil) [reply source/develop/release-management/features/network/ovirt-lldp-labeler.html.md]: Done

File: source/develop/release-management/features/network/ovirt-lldp-labeler.html.md [hunk 47]
```diff
@@ -0,0 +1,156 @@
...
+First time setup:
+
+1) Write username and password into `/etc/ovirt-lldp-labeler/conf.d/ovirt-lldp-credentials.conf`
+2) Write cluster name and API address into `/etc/ovirt-lldp-labeler/conf.d/ovirt-lldp-labeler.conf`
+
+Repeatable invocation:
+
+1) Run the oVirt LLDP labeler by executing `python /usr/share/ovirt-lldp-labeler/ovirt_lldp_labeler_cli.py`
+  
+After the labeler run, the hosts will automatically sets `lldp_vlan_10`
```
[the remark this answers is not in the document]
Comment (almusil) [reply source/develop/release-management/features/network/ovirt-lldp-labeler.html.md]: Done

File: source/develop/release-management/features/network/ovirt-lldp-labeler.html.md [hunk 48]
```diff
@@ -0,0 +1,156 @@
...
+
+1) Write username and password into `/etc/ovirt-lldp-labeler/conf.d/ovirt-lldp-credentials.conf`
+2) Write cluster name and API address into `/etc/ovirt-lldp-labeler/conf.d/ovirt-lldp-labeler.conf`
+
+Repeatable invocation:
+
+1) Run the oVirt LLDP labeler by executing `python /usr/share/ovirt-lldp-labeler/ovirt_lldp_labeler_cli.py`
+  
+After the labeler run, the hosts will automatically sets `lldp_vlan_10` 
+label on top of the `eth1`, which results in the automatic connection of
```
[the remark this answers is not in the document]
Comment (almusil) [reply source/develop/release-management/features/network/ovirt-lldp-labeler.html.md]: Done

File: source/develop/release-management/features/network/ovirt-lldp-labeler.html.md (hunk 40 above)
Comment (almusil) [reply source/develop/release-management/features/network/ovirt-lldp-labeler.html.md]: Done

File: source/develop/release-management/features/network/ovirt-lldp-labeler.html.md (hunk 41 above)
Comment (almusil) [reply source/develop/release-management/features/network/ovirt-lldp-labeler.html.md]: Done

File: source/develop/release-management/features/network/ovirt-lldp-labeler.html.md (hunk 42 above)
Comment (almusil) [reply source/develop/release-management/features/network/ovirt-lldp-labeler.html.md]: Done

File: source/develop/release-management/features/network/ovirt-lldp-labeler.html.md (hunk 43 above)
Comment (almusil) [reply source/develop/release-management/features/network/ovirt-lldp-labeler.html.md]: Done

File: source/develop/release-management/features/network/ovirt-lldp-labeler.html.md (hunk 44 above)
Comment (almusil) [reply source/develop/release-management/features/network/ovirt-lldp-labeler.html.md]: Done

**Pull request merged**